tensorflow/noscope/yolo.h

```diff
@@ -0,0 +1,36 @@
+#include <string>
+#include <atomic>
+
+#include "tensorflow/noscope/darknet/src/yolo.h"
+#include "tensorflow/noscope/darknet/src/image.h"
+
+namespace noscope {
+
+class YOLOClassify{
+
+public:
+  YOLOClassify(size_t resolution,
+               yolo::YOLO *model);
+  ~YOLOClassify();
+
+  //lock the yolo nn, return success
+  bool Lock(int stream_id);

```

Comment:
We don't want to be implementing our own synchronization if possible. There should be open source implementations of synchronous multi-producer/consumer queues we can pull from